config/src/main/java/org/axonframework/config/EventProcessingConfigurer.java

```diff
@@ -823,4 +823,30 @@ static DeadLetteringInvokerConfiguration noOp() {
             return (config, builder) -> builder;
         }
     }
+
+    /**
+     * Register the given {@code deadLetterProvider} as a default to build a {@link SequencedDeadLetterQueue} for
+     * {@link EventProcessor}s created in this configuration.
+     * <p>
+     * The {@code deadLetterProvider} is invoked only if a processing group is configured to have a dlq.
+     *
+     * @param deadLetterProvider a builder {@link Function} that provides a {@link SequencedDeadLetterQueue} for a
+     *                           processing group. It's possible to return null depending on the processing group, if so
+     *                           the next ones will be tried.
+     * @return the current {@link EventProcessingConfigurer} instance, for fluent interfacing
+     */
+    default EventProcessingConfigurer registerDeadLetterProvider(
+            BiFunction<Configuration, String, SequencedDeadLetterQueue<EventMessage<?>>> deadLetterProvider) {
+        return this;
+    }
+
+    /**
+     * Provides a function to get a {@link SequencedDeadLetterQueue} based on the set {@code deadLetterProviders}.
+     *
+     * @param processingGroup the processingGroup to provide a sequnced dead letter queue for
+     * @return a function to provide a {@link SequencedDeadLetterQueue}
+     */
+    default Function<Configuration, SequencedDeadLetterQueue<EventMessage<?>>> provideDlq(String processingGroup) {
```

Comment:
I was not entirely convinced about the method name `provideDlq`, but after seeing the usage in the `AxonAutoConfiguration` class I understand why you choose it.
You use this method to provide the factory `Function` to invoke `registerDeadLetterQueue`.
Hence, it provides the DLQ.

So, does this mean you would expect non-Spring users to invoke this method and pass the result directly to the `registerDeadLetterQueue` too?
Wouldn't it be more straightforward to adjust this method to enable the processing groups for which a default DLQ should be injected?
So, some form of processing group filter (`Predicate`) for which the default DLQ can be set?

Reply:
It might indeed be nicer for non spring users, when setup like that. For Spring users we could keep the same property, and add some logic like only for the processing group in this list, return a sequenced dead letter queue, else null.